multi-arch/samples.md

```diff
@@ -0,0 +1,158 @@
+---
+title: Samples-Non-X86
+authors:
+  - "@gabemontero"
+reviewers:
+  - "@smarterclayton"
+  - "@crawford"
+  - "@bparees"
+  - "@adambkaplan"
+approvers:
+  - "@smarterclayton"
+  - "@crawford"
+  - "@bparees"
+creation-date: 2019-09-20
+last-updated: 2019-09-20
+status: implementable
+see-also:
+replaces:
+superseded-by:
+---
+
+# Sample Imagestream and Templates on non-x86 Platforms
+
+As of this writing, all samples are provided and owned by non-OpenShift 
+Development teams within Red Hat.  And none of those teams 
+
+ - provide host images on registry.redhat.io or quay.io that support PPC or Z 
+ - declare imagestreams in https://github.com/openshift/library that could reference such images
+
+OpenShift is now looking at providing support for the two non-x86 architectures from IBM in v4.x.
+
+Given that, until a resonable set of samples are provided for the IBM non-x86 architectures by either
+those other teams in Red Hat, or by IBM, the samples operator should ensure that it
+
+ - does not install any x86 imagestreams, or templates referencing those imagestreams, on non-x86 systems
+ - and by extension, always provides valid ClusterOperator statuses during install and upgrade, as this amounts to a no-op on those platforms
+
+And in case the implication is not clear, OpenShift development will not be developing or maintaining 
+any non-x86 images for these existing samples.
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift/docs]
+
+## Summary
+
+As part of achieving what is described in the opening section, the samples operator needs
+to achieve the desired results within the context of how multi-arch is being achieved for 
+the entire product.
+
+Alex Crawford and Trevor King have indicated that 
+ - clusters would remain homogenous at least through 4.5/4.6 at the time of this writing
+ - use of golang's runtime.GOARCH is sufficient for determining one's architecture
+
+That said, the samples operator config object currently has a []string array defined "Architectures"
+in anticipation of heterogenous clusters.  If that requirement ever gets into plan, the current understanding
+after talking to Alex and Trevor are
+ - some sort of list of architectures in a cluster would be provided in the infrastructure config
+ - k8s styled selectors of some sort will be provided/utilized to pin pod leveraging non-x86 images to the required nodes; concerns for this pinning thus would not fall on the samples operator
```

Comment:
though we would have to make a decision about which imagestreams/templates to install (e.g. the x86 ones or the ppc ones, on a cluster that has both x86 + ppc nodes.  We likely can't install both because of naming collisions.  So we'd probably have to pick one and it's up to the admin to reconfigure things later if they want the other...  but we can cross that bridge when we come to it)

Reply:
The alternative is to dive head-first into supporting manifest lists with ImageStreams - that would truly be a v2 change.